app/routes.py

```diff
@@ -1,2 +1,338 @@
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+    }
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods= ["PATCH"])
+
+
+
+def mark_incomplete(task_id):
+
+    task = Task.query.get(task_id)
+    # taskvariable = None
+    if task is None:
+        return jsonify(None), 404
+
+    task.completed_at = None
+    
+
+    db.session.commit()
+
+    return {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+    }
+
+
+# goals_bp = Blueprint("goals_bp", __name__, url_prefix=("/goals"))
+@goals_bp.route("", methods=["GET", "POST"])
+#any time we make a request to our api and call localhost5000, we are going to run the function.
+#what will run depends on if the client calls GET or POST
+def handle_goals(): #view function names should be unique
+    if request.method == "GET":
+        goals = Goal.query.all() #Task is a child of SQLA so we can use the query method, 
+        #query is method from sqla, paired with .all, retrieves all rows from task table in database, from there SQLA will turn all those rows into instances
+        #of the task class, and all method will return all instances of the Task class into a list, and that list is assigned to variable task
+        #Task.query.all.sort?? we can order the result the result of our selection
+        goals_response = []
+        for goal in goals: #<Task 2342> - returns object, if we want to see attributes we have to 
+            #access the attributes. So we break down objhects in that list in dictionary so we can see each instance's attributes
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+        return jsonify(goals_response)
+        #put response in JSON format to be used by rest of Flask, include headers, etc
+        #help turn raw data into response that Flask can use
+        #Flask develops web applications and includes features that helps you interface more easily with interfaces and defining APIs, ie glue code that helps our app
+        #application that work with other libraries that work with website
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            # return make_response("400 Bad Request")
+            return {"details": "Invalid data"}, 400
+
+        new_goal = Goal(
+            title= request_body["title"]
+        )
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }, 201
+
+@goals_bp.route("/<goal_id>", methods= ["GET", "PUT", "DELETE"])
```

Comment:
We can also separate each request method into different functions for this route as well.